### PR TITLE
Initial ci-operator config for openshift/master-dns-operator

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

**Comment by petr-muller** [comment]
Also, just to be sure: most repos make use of the `approve` plugin by adding a little stanza to /cluster/ci/config/prow/plugins.yaml, like ([here](https://github.com/openshift/release/pull/2282/files#diff-5da33c7edc88b9ba312e0e1eda240a8fR359)). Is it intentional for openshift/master-dns-operator to not use it?

**Comment by csrwng** [comment]
@petr-muller thank you for taking a look at this so quickly. I've now added the proper OWNERS files. The reason that we don't use the approve plugin is that for this repo just like hive, we have a small team where every member is an approver, the extra step of approving PRs seems like overkill.

File: ci-operator/config/openshift/master-dns-operator/openshift-master-dns-operator-master.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    cluster: https://api.ci.openshift.org
+    name: origin-v4.0
+    namespace: openshift
+    tag: base
+binary_build_commands: make build
+canonical_go_repository: github.com/openshift/master-dns-operator
+images:
+- dockerfile_path: Dockerfile
+  from: base
+  to: master-dns-operator
```
Comment (smarterclayton) [inline ci-operator/config/openshift/master-dns-operator/openshift-master-dns-operator-master.yaml]: We aren’t using the word “master” if we can avoid it.

Comment (smarterclayton) [reply ci-operator/config/openshift/master-dns-operator/openshift-master-dns-operator-master.yaml]: I don’t really think this is the right name for this.  Its too specific and confusing.

What other code will this live in?  Why isn’t this part of openshift-kube-apiserver-operator?

Comment (smarterclayton) [reply ci-operator/config/openshift/master-dns-operator/openshift-master-dns-operator-master.yaml]: If this is about etcd management then give a name like that and let’s fold the bootstrap aspects of setup-etcd-bootstrap into this or cluster-bootstrap.

Comment (csrwng) [reply ci-operator/config/openshift/master-dns-operator/openshift-master-dns-operator-master.yaml]: @smarterclayton We can definitely rename, though not sure that it makes sense to lump with the apiserver operator. This deals with cloud infrastructure and not just control plane stuff. The other cloud infra operator we have is the machine api operator but not sure it fits there either. How does `control-plane-dns-operator` sound?

Comment (csrwng) [reply ci-operator/config/openshift/master-dns-operator/openshift-master-dns-operator-master.yaml]: Ahh missed your last comment. Ack. Makes sense

**Pull request closed without merging**